### PR TITLE
Add GH workflow to trigger release candidate

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Description
Part of #3032 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: .github/workflows/release-candidate.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Release candidate
+
+on:
+  workflow_dispatch: {}
```
Comment (aclevername) [inline .github/workflows/release-candidate.yaml:4]: this means its a manual trigger right?

Comment (michaelbeaumont) [reply .github/workflows/release-candidate.yaml:4]: yep

File: build/scripts/tag-release-candidate.sh [hunk 2]
```diff
@@ -63,21 +70,21 @@ git tag --annotate --message "${m}" --force "latest_release"
...
 # Increase next development iteration if needed
-if [ "${master_version}" == "${candidate_for}" ]; then
+if [ "${dev_version}" == "${candidate_for}" ]; then
   echo "Preparing for next development iteration"
   release_generate development
   git add ./pkg/version/release.go
   git commit --message "Prepare for next development iteration"
-  git push origin master:master
+  git push origin "${default_branch}":"${default_branch}"
 fi
```
Comment (aclevername) [inline build/scripts/tag-release-candidate.sh:90]: this should work as admins can push (I disabled it when releasing during the week), but we should remember a follow up PR to change this to open a PR instead IMO

**Review by aclevername — APPROVED** [review]
LGTM, I don't fully understand all of the github action syntax, but we won't know it works until we try it!

File: .github/workflows/release-candidate.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+jobs:
+  rc:
+    name: Trigger release candidate build
+    runs-on: ubuntu-latest
+    container: weaveworks/eksctl-build:6bfc53c9862b0c44697dddd14df9e571eec0cdfa
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.WEAVEWORKSBOT_TOKEN }}
```
Comment (Callisto13) [inline .github/workflows/release-candidate.yaml:15]: this needed?

File: build/scripts/tag-release-candidate.sh (hunk 2 above)
Comment (michaelbeaumont) [reply build/scripts/tag-release-candidate.sh:90]: this may fail actually...

Comment (michaelbeaumont) [reply build/scripts/tag-release-candidate.sh]: @aclevername  see newest commit

Comment (aclevername) [reply build/scripts/tag-release-candidate.sh]: nice!

**Comment by michaelbeaumont** [comment]
@aclevername I just added the release branch step to the script, not sure why it wasn't being done automatically before...

**Pull request merged**